packages/flutter/lib/src/physics/spring_simulation.dart

```diff
@@ -248,35 +252,33 @@ class _UnderdampedSolution implements _SpringSolution {
     double distance,
     double velocity,
   ) {
-    final double w = math.sqrt(4.0 * spring.mass * spring.stiffness - spring.damping * spring.damping) /
-        (2.0 * spring.mass);
+    final double w =
+        math.sqrt(4.0 * spring.mass * spring.stiffness - spring.damping * spring.damping) / (2.0 * spring.mass);
     final double r = -(spring.damping / 2.0 * spring.mass);
     final double c1 = distance;
     final double c2 = (velocity - r * distance) / w;
     return _UnderdampedSolution.withArgs(w, r, c1, c2);
   }
 
   _UnderdampedSolution.withArgs(double w, double r, double c1, double c2)
-    : _w = w,
-      _r = r,
-      _c1 = c1,
-      _c2 = c2;
+      : _w = w,
+        _r = r,
+        _c1 = c1,
+        _c2 = c2;
 
   final double _w, _r, _c1, _c2;
 
   @override
   double x(double time) {
-    return (math.pow(math.e, _r * time) as double) *
-           (_c1 * math.cos(_w * time) + _c2 * math.sin(_w * time));
+    return (math.pow(math.e, _r * time) as double) * (_c1 * math.cos(_w * time) + _c2 * math.sin(_w * time));
   }
 
   @override
   double dx(double time) {
     final double power = math.pow(math.e, _r * time) as double;
     final double cosine = math.cos(_w * time);
     final double sine = math.sin(_w * time);
-    return power * (_c2 * _w * cosine - _c1 * _w * sine) +
-           _r * power * (_c2 *      sine   + _c1 *      cosine);
+    return power * (_c2 * _w * cosine - _c1 * _w * sine) + _r * power * (_c2 * sine + _c1 * cosine);
```

Comment:
this one got messed up at some point but again it would ideally be aligned for clarity.

the one above this (line 273) could similarly be improved.

Reply:
Yeah, this one would have to be opted-out, I think. Can't think of a good way to express this in a formatter compatible way that would keep alignment.

```dart
    return     power * (_c2 * _w * cosine - _c1 * _w * sine) +
          _r * power * (_c2 *      sine   + _c1 *      cosine);
```